### PR TITLE
clean up pretext build

### PR DESCRIPTION
**Pull request by oscarlevin**
Some initial work, in case anyone is curious where this is headed.

### CODE REVIEW

File: pretext/project/__init__.py [hunk 1]
```diff
@@ -342,24 +339,60 @@ def build(
...
+                    pub_file=self.publication_abspath().as_posix(),
+                    stringparams=self.stringparams,
+                    xmlid_root=xmlid,
+                    file_format="html",
+                    extra_xsl=custom_xsl,
+                    out_file=None,
+                    dest_dir=self.output_dir_abspath().as_posix(),
+                )
+                # For codechat:
+                project_path = utils.project_path(self.source_abspath())
```
Comment (oscarlevin) [inline pretext/project/__init__.py:354]: @bjones1, could you look over the next few lines to make sure this is doing what it needs to?  I wasn't sure what the best way to do the project_path in the refactor (might it just be `Path().resolve()`?)

File: pretext/project/__init__.py [hunk 2]
```diff
@@ -341,25 +338,60 @@ def build(
...
+                    pub_file=self.publication_abspath().as_posix(),
+                    stringparams=self.stringparams,
+                    xmlid_root=xmlid,
+                    file_format=self.compression or "html",
+                    extra_xsl=custom_xsl,
+                    out_file=None,
+                    dest_dir=self.output_dir_abspath().as_posix(),
+                )
+                # For codechat:
+                project_path = utils.project_path(self.source_abspath())
```
Comment (bjones1) [inline pretext/project/__init__.py:353]: We should remove `utils.project_path` and replace it with `Project.abspath()`.

File: pretext/project/__init__.py [hunk 3]
```diff
@@ -341,25 +338,60 @@ def build(
...
+                assert (
+                    project_path is not None
+                ), f"Invalid project path to {self.source_abspath()}."
+                codechat.map_path_to_xml_id(
+                    self.source_abspath(), project_path, self.output_dir_abspath().as_posix()
```
Comment (bjones1) [inline pretext/project/__init__.py:358]: Drop `as_posix()` here -- it's not necessary.

File: pretext/project/__init__.py [hunk 4]
```diff
@@ -341,25 +338,60 @@ def build(
...
-            )
+            if self.format == "html":
+                core.html(
+                    xml=self.source_abspath(),
+                    pub_file=self.publication_abspath().as_posix(),
+                    stringparams=self.stringparams,
+                    xmlid_root=xmlid,
+                    file_format=self.compression or "html",
+                    extra_xsl=custom_xsl,
+                    out_file=None,
```
Comment (bjones1) [inline pretext/project/__init__.py:349]: If we're making a compressed HTML file, shouldn't we allow the `output_file` to determine the name of the resulting .zip archive?

File: pretext/project/__init__.py [hunk 5]
```diff
@@ -341,25 +338,60 @@ def build(
...
-                custom_xsl=custom_xsl,
-                xmlid=xmlid,
-                zipped=self.compression is not None,
-                project_path=self._project.abspath(),
-                latex_engine=self.latex_engine,
-                executables=self._project._executables.dict(),
-                # TODO: what if this isn't defined? Should we have a default in the project file instead?
-                braille_mode=self.braille_mode,
-            )
+            if self.format == "html":
```
Comment (bjones1) [inline pretext/project/__init__.py:341]: I would use `if self.format == Format.HTML:` instead.

File: pretext/project/__init__.py (hunk 4 above)
Comment (oscarlevin) [reply pretext/project/__init__.py:349]: I'm not sure what happens in core if both a `dest_dir` and `out_file` are specified.  I suppose that isn't our problem though.

File: pretext/project/__init__.py (hunk 3 above)
Comment (oscarlevin) [reply pretext/project/__init__.py:358]: Something is strange here.  without the `.as_posix()` I get complaints: Argument of type "Path" cannot be assigned to parametry "des_dir" of type "str" in function "map_path_to_xml_id"

File: pretext/project/__init__.py (hunk 4 above)
Comment (oscarlevin) [reply pretext/project/__init__.py:349]: Also, the `@validator` for `output_filename` would need to change in the case that compression is set.

Comment (oscarlevin) [reply pretext/project/__init__.py]: This should be fixed now.  Please take a look, @bjones1

**Comment by oscarlevin** [comment]
@StevenClontz: I added in the custom build to the refactor, but didn't do anything with the output filename or directory.  Can you double check that my implementation does what you expect?

**Comment by oscarlevin** [comment]
I'm going to merge this so I can add it to work I'm doing on `generate`.  But @StevenClontz, when you get a chance, please double check the `custom` build logic.

**Pull request merged**